src/context/ColonyContext.tsx

```diff
@@ -43,13 +60,50 @@ export const ColonyContextProvider = ({
     fetchPolicy: 'cache-and-network',
   });
 
-  const colony = data?.getColonyByName?.items?.[0] ?? undefined;
+  const updateColony = useCallback(() => {
```

Comment:
I think we should take a step back - is the intention to just refetch the colony query following the watch/unwatch mutation so that the UI updates with the newest data?

If so, there are two solutions already provided that Apollo that require minimal/no changes to ColonyContext:
- On the mutation itself, you can specify the `refetchQueries` field to include the colony query:
```js
const [watch, { data: watchState, loading: loadingWatch }] =
    useCreateWatchedColoniesMutation({
      variables: {
        input: {
          colonyID: colony?.colonyAddress || '',
          userID: user?.walletAddress || '',
        },
       refetchQueries: [GetFullColonyByNameDocument]
      },
    });
```
The `GetFullColonyByNameDocument` is generated automatically by codegen and exported from `~gql`.
- `useGetFullyColonyByNameQuery` also returns a `refetch` function, which you can expose to the context consumers

That way you shouldn't have to replicate all the existing logic

Reply:
Very nice! Works great.